book/src/css.md

```diff
@@ -0,0 +1,264 @@
+Since so many people are already familiar with web development, it makes sense that GTK supports its own variant of CSS.
+
+> We will not explain every piece of CSS syntax used in this chapter.
+> If you are new to it or just need a refresher, have a look at the [MDN Web Docs](https://developer.mozilla.org/en-US/docs/Web/CSS/Syntax)
+
+Let us say we have a button and we want to set its font color to red.
+Every widget has a corresponding CSS node.
+In the case of `gtk::Button`, this node is called `button`.
+Therefore, we create a `style.css` file with the following content:
+
+<span class="filename">Filename: listings/css/1/style.css</span>
+
+```css
+{{#rustdoc_include ../listings/css/1/style.css}}
+```
+
+Next, we load the CSS file in the startup step of the application.
+As usual, the widgets are created during the activate step.
+
+<span class="filename">Filename: listings/css/1/main.rs</span>
+
+```rust ,no_run,noplayground
+{{#rustdoc_include ../listings/css/1/main.rs:main}}
+```
+
+When we now run the app, we notice that our button *and* the "close" button are red.
+Probably not what we wanted, but that is what we typed.
+We did not specify for which button the rule should apply, so it was applied to both of them.
+
+<div style="text-align:center"><img src="img/css_1.png"/></div>
+
+>The `GtkInspector` comes in quite handy (not only) when playing with CSS.
+>Make sure that the window of your app is focused and press <kbd>Ctrl</kbd> + <kbd>Shift</kbd> + <kbd>D</kbd>.
+> A window will pop up which lets you browse and even manipulate the state of your app.
+> Open the `CSS` view and override the button color with the following snippet.
+>```css
+>button {
+>  color: blue;
+>}
+>```
+
+## Classes
+
+[Class selectors](https://developer.mozilla.org/en-US/docs/Web/CSS/Class_selectors) are one way to limit the amount of elements a CSS rule applies to.
+Let us look at few different scenarios where they are involved.
+
+
+### Classes Applied by GTK
+
+GTK adds style classes to many of its nodes, often depending on a certain condition.
+In the case of the node [`button`](https://gtk-rs.org/gtk4-rs/stable/latest/docs/gtk4/struct.Button.html#css-nodes), the style class `.text-button` will be added when the content of the corresponding widget is just a label.
+That is why we create a new CSS rule which only applies to `button` nodes with the style class `text_button`.
+
+
+<span class="filename">Filename: listings/css/2/style.css</span>
+
+```css
+{{#rustdoc_include ../listings/css/2/style.css}}
+```
+
+<div style="text-align:center"><img src="img/css_2.png"/></div>
+
+### Classes Applied Manually
+
+If we want to apply different rules to nodes which got the same style classes added, we will have to think of a different strategy.
+For that we add our own CSS class with [`add_css_class`](../docs/gtk4/prelude/trait.WidgetExt.html#tymethod.add_css_class).
```

Comment:
```suggestion
If you need to style differently two widgets that have the same automatic GTK style classes, you can add your own CSS class using [`add_css_class()`](../docs/gtk4/prelude/trait.WidgetExt.html#tymethod.add_css_class).
```

Reply:
What is an automatic GTK style class in this context?

Reply:
Title of the section right before about stuff like `.text-content`

Reply:
`.text-button` you mean? Sure but there's `add your own CSS class using add_css_class()` in the same sentence. :)